### ISSUE TITLE
AUTH: Received control message: AUTH_FAILED

### ISSUE THREAD

**Issue by Omaha2002** [edgd1er/nordvpn-proxy#37]
I've created a nordvpn_creds file with the Nordvpn credentials although the readme says to create two files with the login and pass.. the instructions for me are a bit confusing.

nordvpn_creds containing:

nordvpn_login=******
nordvpn_pass=*****

log:

proxy-1  | 2024-01-21 11:01:20: OVPN: CREATING TUN INTERFACE /DEV/NET/TUN: 
proxy-1  | 2024-01-21 11:01:20: INFO: NORDVPN: Using cached values for json_countries
proxy-1  | 2024-01-21 11:01:20: INFO: NORDVPN: Using cached values for json_groups
proxy-1  | 2024-01-21 11:01:20: INFO: NORDVPN: Using cached values for json_technologies
proxy-1  | 2024-01-21 11:01:20: INFO: Checking curl installation
proxy-1  | 2024-01-21 11:01:20: INFO: DNS: resolution ok
proxy-1  | 2024-01-21 11:01:21: INFO: OVPN: ok, configurations download site reachable
proxy-1  | 2024-01-21 11:01:21: INFO: NORDVPN: Removing existing configs in /config
proxy-1  | 2024-01-21 11:01:21: INFO: Checking NORDPVN API responses
proxy-1  | 2024-01-21 11:01:21: INFO: Selecting the best server...
proxy-1  | 2024-01-21 11:01:21: INFO: NORDVPN: Searching for country : xxxxxxxx (153)
proxy-1  | 2024-01-21 11:01:21: INFO: NORDVPN, searching for group: legacy_p2p
proxy-1  | 2024-01-21 11:01:21: INFO: NORDVPN: searching for technology: openvpn_udp
proxy-1  | 2024-01-21 11:01:32: INFO: OVPN: Best server : xxxx.nordvpn.com, load: 9
proxy-1  | 2024-01-21 11:01:32: INFO: NORDVPN: selected: xxxx.nordvpn.com, VPN_PROVIDER_HOME: /config
proxy-1  | 2024-01-21 11:01:32: INFO: OVPN: Downloading config: xxxx.nordvpn.com.ovpn
proxy-1  | 2024-01-21 11:01:32: INFO: OVPN: Downloading from: https://downloads.nordcdn.com/configs/files/ovpn_udp/servers/xxxx.nordvpn.com.udp.ovpn
proxy-1  | 2024-01-21 11:01:32: INFO: OVPN: ########################################################
proxy-1  | 2024-01-21 11:01:32: INFO: OVPN: Files found: 1
proxy-1  | 2024-01-21 11:01:32: INFO: OVPN: config: /config/xxxx.nordvpn.com.ovpn
proxy-1  | 2024-01-21 11:01:32: INFO: OVPN: openvpn log level: 0
proxy-1  | 2024-01-21 11:01:32: INFO: OVPN: ########################################################
proxy-1  | 2024-01-21 11:01:32: OVPN: FOUND EXISTING OPENVPN CREDENTIALS AT /CONFIG/OPENVPN-CREDENTIALS.TXT: 
proxy-1  | 2024-01-21 11:01:32: OVPN: SETTING OPENVPN CREDENTIALS...: 
proxy-1  | 2024-01-21 11:01:32: OVPN: LOCAL_NETWORK: 192.168.6.0/24, GATEWAY: 172.19.0.1, DEVICE ETH0: 
proxy-1  | adding route to local network 192.168.6.0/24 via 172.19.0.1 dev eth0
proxy-1  | 2024-01-21 11:01:32 NOTE: the current --script-security setting may allow this configuration to call user-defined scripts
proxy-1  | 2024-01-21 11:01:32 WARNING: this configuration may cache passwords in memory -- use the auth-nocache option to prevent this
proxy-1  | 2024-01-21 11:01:33 AUTH: Received control message: AUTH_FAILED
proxy-1  | 2024-01-21 11:01:33: OVPN: VPN CONNECTION LOST, STOPPING UNBOUND: 
proxy-1  | 2024-01-21 11:01:33: OVPN: WAITING 30 SECONDS BEFORE RESTARTING:

**Comment by edgd1er** [comment]
you are refering to `create file nordvpn_login containing your login, nordvpn_pass containing your password`

You are right, the readme is not clear enough about the expected structure. first line is the login, second line is the password.
the file should only contain the value.
nordvpn_creds:
```
thisismyusername
thisismypassword
```
may I ask how You would rephrase that so it's more clear ?

**Comment by Omaha2002** [comment]
Thank you for the quick response, I think the description you wrote here is clear enough.

I did create the nordvpn_creds with the proper values but still getting:

2024-01-21 17:33:20 AUTH: Received control message: AUTH_FAILED

**Comment by Omaha2002** [comment]
btw in your README it says:  create file **nordvpn_login** containing your login, nordvpn_pass containing your password.

Shouldnt that read "create file **nordvpn_creds** containing your login and password.

**Comment by Omaha2002** [comment]
ok, figured it out, the credentials shouldnt be the ones you login to NordVPN but the ones you can create:

1. Go to NordVPN website and log in
2. Under accounts - Services - click NordVPN
3. Click - Set up NordVPN manually - at the bottom right of the page.
4. You will receive a verification code in your email that you use for NordVPN services. Type the code in the popup window the preceded the email check.
5. Copy the credentials using the “Copy” buttons on the right for your new encrypted user name and password and paste them in nordvpn_creds

**Comment by edgd1er** [comment]
that's right. As I'm not using anymore that image, as I use a new one with nordvpn client: [nordlynx-proxy](https://hub.docker.com/r/edgd1er/nordlynx-proxy). openvpn and wireguard protocols are available.

I experimented better upload/download speed with nordvpn's wireguard implementation: nordlynx.

I will add information about how to get credentials. I think, you will recognize the wording ;)

**Comment by edgd1er** [comment]
Please feel free to reopen if needed
